Investigate freezing test

mikeu opened the pull request:
Ignore this PR for now.

The tests all run fine locally, including the quarantined one, so I'm opening a PR to incrementally add to the test and run it on CI to see where it starts misbehaving.

mikeu commented:
@DonNicoJs so it seems that whatever is causing the quarantined test to hang in CI only happens when the same prop is modified twice via `setProps`, without awaiting the first change.

I'll be honest, I don't fully understand what exactly this test needs to accomplish, or how it relates to animations—is updating the props more than once _without_ awaiting the changes on the vm critical to the logic? Or is it OK if the "short time period" that the test mentions during which the props are updated multiple times actually spans a few vm ticks?

The current version of the test in this PR is passing in CI without any hangs or other problems, and it includes the same sequence of events as the original, with the same expectations at the end. The only difference is that there is a vm tick awaited each time the props change, instead of it happening all at once.

---

I looked a little bit into why this might be causing problems, but haven't found anything concrete yet. The tests all pass locally for me (and, I'm guessing, for you?). The [vue-test-utils docs](https://vue-test-utils.vuejs.org/guides/#writing-asynchronous-tests-new) say that "you **must** wait for updates to run after you change a reactive property." I generally that used to mean "...if you want to see the change reflected in the wrapper," but maybe it actually means "if you want the test to function correctly (in all environments)."

DonNicoJs commented:
@mikeu thank you for investigating this, I start to think that this test is not helping us at all :) WDYT? should we just drop it? I am afraid is giving us confidence that what is testing works but it does not actually.

mikeu commented:
> @mikeu thank you for investigating this, I start to think that this test is not helping us at all :) WDYT? should we just drop it? I am afraid is giving us confidence that what is testing works but it does not actually.

@DonNicoJs I tend to agree with you, I think the three subsequent tests about no-blocking-animations for the centre, zoom, and bounds do a better job of testing the expected behaviour of this library. Verifying that the map object updates its values does _feel_ worth doing at least at first glance, but that's just testing Leaflet in the end.

Also, the test appears to behave identically whether the wrapper is created with `noBlockingAnimations` set to true, false, null, undefined, or if the key is removed entirely from the options object. So whatever it is testing, I don't think it's the animation behaviour.

My main worry about dropping it is the comment that it is the "Most important test for no-blocking-animations" although that is followed up with "However, I suspect animations are never triggered in unit tests" which again would mean that very little useful, and nothing of Vue2Leaflet itself, was actually getting tested. Was this a test you added by any chance?

I'll push one more commit here removing the test, so just merge that if you're happy with that decision. Otherwise we can keep investigation the issue, including...

> Also BTW the tests hang for me locally! What version of node are you running them? Did you update the local packages?

I am currently running the tests under node v12.18.2, with the following summary:
```
Test Suites: 16 passed, 16 total
Tests:       3 skipped, 258 passed, 261 total
Snapshots:   0 total
Time:        3.167s
Ran all test suites.
Done in 3.75s.
```

I'll now run `npm install`... OK, apparently I haven't done that in a while :)
```
added 493 packages from 180 contributors, removed 206 packages, updated 298 packages and audited 2249 packages
```
...and now the test hangs locally for me as well! So I guess something in an update to one of the dependencies has changed the relevant behaviour. I have the output of `npm ls` before and after that install, so let me know if you'd like to start digging into this further. But at this point I think I'd be happy to just merge the removal of that test.

DonNicoJs commented:
@mikeu Merged, I don't think we should invest more time on this!
